content/snapshot-resource/index.md

```diff
@@ -0,0 +1,261 @@
+with:
+
+```elm
+collect log
+```
+
+And compute the snapshot at size 5 by slicing the log to that size:
+
+```elm
+collect (take 5 log)
+```
+***
+
+Let's redefine “record” as the combination of the data with the entry key.
+
+```elm
+type Record =
+  { key : ID
+  , data : Dict Name Value
+  }
+```
+
+Note that the attribute `data` holds the same type of data structure as the
+blob. The resason for not using the blob here is because this opens the door
+to return data that is not a perfect match with the blob. For example, the RFC
+draft for removing fields would benefit from this flexibility.
+
+Note that `Value` is either a string or a set of strings, the same as the blob
+definition.
+
+The operation to compute the record for a given entry can be defined as:
+
+```elm
+view : BlobStore -> Entry -> Result Error Record
+```
+
+Where `BlobStore` is the complete set of known blobs. Given that the blob
+store may not contain the blob referenced by the entry, this computation can
+fail with an error.
+
+***
+**EXAMPLE:**
+
+For example, to compute the latest set of records for a given log:
+
+```elm
+map (view store) (collect log)
+```
+
+So, first we compute the snapshot for the full log and then we transform each
+entry into a record.
+***
+
+### REST API
+
+#### Get a snapshot
+
+Returns an array of entry resources.
+
+This resource MAY be paginated like the rest of the collections.
+
+***
+#### Endpoint
+
+```
+GET /snapshots/{log-size}
+```
+
+#### Parameters
+
+|Name|Type|Description|
+|-|-|-|
+|`log-size`|Integer|The log size to compute the snapshot.|
+
+Note: We can consider some kind of alias for the latest log size like the
+`latest` tag used by Docker.
+***
+
+
+#### Get a snapshot entry by key
+
+Returns an entry resource.
+
+***
+#### Endpoint
+
+```
+GET /snapshots/{log-size}/{key}
+```
+
+#### Parameters
+
+|Name|Type|Description|
+|-|-|-|
+|`log-size`|Integer|The log size to compute the snapshot.|
+|`key`|ID|The unique identifier for an element of the register.|
+***
+
+#### Get a record by id
+
+Returns a record resource.
+
+***
+#### Endpoint
+
+```
+GET /records/{id}
+```
+
+#### Parameters
+
+|Name|Type|Description|
+|-|-|-|
+|`id`|ID|The unique identifier for an element of the register.|
+***
+
+##### JSON serialisation
+
+The record resource MUST be a JSON object with the data for the record. It
+MUST also have the key of the entry using the reserved attribute `_id`. Given
+that an attribute name is of type `Name`, it is not possible to have a clash
+with `_id` (underscores (`_`) are not allowed as the first character for a
+type `Name`).
+
+***
+**EXAMPLE:**
+
+For example, the most recent record for the contry identified by `GB`:
+
+```http
+GET /records/GB HTTP/1.1
+Accept: application/json
+Host: country.register.gov.uk
+```
+
+```http
+{
+  "_id": "GB",
+  "name": "United Kingdom",
+  "official-name": "The United Kingdom of Great Britain and Northern Ireland",
+  "citizen-names": ["Briton", "British citizen"]
```

Comment:
this would still have the current register key convention in our usage as well as `_id` right? i.e. `"country": "GB"`

Reply:
It would still have the field `country` because it is data part of the blob